app/src/hooks/useEnvironment.tsx

```diff
@@ -1,9 +1,8 @@
-import { Environment } from '@/store/environmentStore'
-import { shouldUseTestnet } from '@/utils/env'
+import { useEnvironmentStore } from '@/store/environmentStore'
 
 const useEnvironment = () => {
-  const environment = shouldUseTestnet ? Environment.Testnet : Environment.Mainnet
-  return environment
+  const current = useEnvironmentStore(state => state.current)
+  return current
```

Comment:
```suggestion
  return useEnvironmentStore(state => state.current)
```

if this works, it's cleaner imo